aws/structure.go

```diff
@@ -2754,6 +2754,7 @@ func flattenIoTRuleFirehoseActions(actions []*iot.Action) []map[string]interface
 		if v != nil {
 			result["role_arn"] = *v.RoleArn
 			result["delivery_stream_name"] = *v.DeliveryStreamName
+			result["separator"] = *v.Separator
```

Comment:
The API will omit this value in the response if parameter is not provided in the configuration and cause a panic:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x2e9cb09]

goroutine 406 [running]:
github.com/terraform-providers/terraform-provider-aws/aws.flattenIoTRuleFirehoseActions(0xc0000c9058, 0x1, 0x1, 0x379df40, 0xc000669520, 0x0)
	/Users/bflad/go/src/github.com/terraform-providers/terraform-provider-aws/aws/structure.go:2815 +0x1e9
```

We should prefer to use the AWS SDK provided functions to dereference pointers instead, e.g. `result["separator"] = aws.StringValue(v.Separator)`